### DISCUSSION TITLE
Setting up the cache-ispn.xml file such that no clustering is attempted

### DISCUSSION THREAD

**Discussion by ncphins** [keycloak/keycloak#30897] [Q&A] [answered]
We have multiple Keycloak test servers running on our network. Up until now we have not seen an issue with the clustering, but we are starting to run into version issues. 

@pruivo - You requested I start this discussion and ping you so to get some guidance on setting up `cache-ispn.xml` to avoid clustering and run each of our instances independently. We have not made any changes to the XML file at all. 

I really appreciate the help!

**Comment by pruivo** [comment, marked as answer]
The default `cache-ispn.xml` shipped with Keycloak uses IP multicast for discovery. The trick is to set a different multicast group for each cluster. You can do it by changing the port:

`JAVA_OPTS_APPEND="-Djgroups.mcast_port=46655" ./bin/kc.sh start ...`

As an example, if you start

* node1 with `JAVA_OPTS_APPEND="-Djgroups.mcast_port=46655" ./bin/kc.sh start ...`
* node2 with `JAVA_OPTS_APPEND="-Djgroups.mcast_port=46655" ./bin/kc.sh start ...`
* node3 with `JAVA_OPTS_APPEND="-Djgroups.mcast_port=46656" ./bin/kc.sh start ...`

Both node1 and node2 will belong to the same cluster (port 46655) while node3 will be on its own (port 46656). 

Let me know if I was clear :+1:

**Reply by ncphins** [reply]
I can verify that this allowed us to run the instance with the new version without any issues. I have to work through the best way to set this since it seems like it is overriding a lot of the default opts from the kc.bat. I am seeing this message. But that is something I can work through.

"JAVA_OPTS already set in environment; overriding default settings with values: -Djgroups.mcast_port=46656"

Thank you!

**Reply by pruivo** [reply]
My bad should have suggested `JAVA_OPTS_APPEND` instead.

**Reply by ncphins** [reply]
I was just coming over to update my reply. After looking through the script I noticed that and changed it. Should have done that before responding. Working well now.
